src/transactions/saga.ts

```diff
@@ -222,7 +227,65 @@ function* watchAddressToE164PhoneNumberUpdate() {
   )
 }
 
+function* watchPendingTransaction(transaction: StandbyTransaction) {
+  const { transactionHash, networkId } = transaction
+
+  if (!transactionHash) {
+    Logger.warn(TAG, `Transaction ${transaction.context.id} has no hash, skipping`)
+    return
+  }
+
+  // TODO: Remove this constraint when our viem client supports other networks
+  const supportedNetworkdIds = [NetworkId['celo-mainnet'], NetworkId['celo-alfajores']]
+  if (!supportedNetworkdIds.includes(networkId)) {
+    Logger.warn(TAG, `Transaction ${transaction.context.id} is not on the celo network, skipping`)
+    return
+  }
+
+  try {
+    const receipt = yield* call(publicClient.celo.getTransactionReceipt, {
+      hash: transactionHash as Hash,
+    })
+
+    if (receipt) {
+      if (receipt.status == 'success') {
+        yield* put(
+          transactionConfirmed(transaction.context.id, {
+            transactionHash: receipt.transactionHash,
+            block: receipt.blockNumber.toString(),
+            status: true,
+          })
+        )
+      }
+
+      if (receipt.status == 'reverted') {
+        yield* put(transactionFailed(transaction.context.id))
+      }
+    }
+  } catch (e) {
+    Logger.error(
+      TAG,
+      `Found error when trying to fetch status for transaction with hash: ${transactionHash}`,
+      e
+    )
+    return
+  }
+}
+
+export function* watchPendingTransactions() {
+  while (true) {
+    const pendingStandbyTransactions: StandbyTransaction[] = yield* select(
+      watchablePendingTransactionSelector
+    )
+    for (const transaction of pendingStandbyTransactions) {
+      yield* fork(watchPendingTransaction, transaction)
```

Comment:
nit: we're not really watching the pending tx here, we're just getting the tx receipt right? maybe `getTransactionReceipt` will do?

Reply:
Hmm, what would you mean by "watching"? `getTransactionReceipt` sounds like it should return the receipt of the tx without any side effects, but that's not the case.
I'll change the name anyways, but I'm not sure if it makes a difference

Reply:
i was thinking of "watching" like it is actively watching / waiting / listening for the transaction receipt. but if i understand correctly, we only use the `getTransactionReceipt` RPC method to see if one exists. this method doesn't wait until one exists, it just makes a fetch and acts accordingly to the result.